libnetwork/config/config.go

```diff
@@ -34,38 +34,40 @@ type Config struct {
 	PluginGetter           plugingetter.PluginGetter
 }
 
-// LoadDefaultScopes loads default scope configs for scopes which
-// doesn't have explicit user specified configs.
-func (c *Config) LoadDefaultScopes(dataDir string) {
-	for k, v := range datastore.DefaultScopes(dataDir) {
-		if _, ok := c.Scopes[k]; !ok {
-			c.Scopes[k] = v
-		}
-	}
-}
-
 // ParseConfigOptions parses the configuration options and returns
 // a reference to the corresponding Config structure
 func ParseConfigOptions(opts ...Option) *Config {
 	cfg := &Config{
 		DriverCfg: make(map[string]interface{}),
 		Scopes:    make(map[string]*datastore.ScopeCfg),
 	}
+
+	opts = append(opts, OptionDefaultScopes(cfg.DataDir))
```

Comment:
The value of `cfg.DataDir` is captured before the options are applied, so will always be the zero value. A user-configured value using `OptionDataDir()` would not be respected.

Reply:
🤦 you are so right; good catch. I started with inlining the code, then thought it would be nice to keep the GoDoc / description, so I went back and changed it to a `Option`, but completely overlooked this.

As this one is only used internally, I should probably also un-export it, and remove the closure.

Reply:
@corhere updated; PTAL